src/server/scripts/Commands/cs_cast.cpp

```diff
@@ -194,9 +204,8 @@ class cast_commandscript : public CommandScript
         Unit* target = handler->getSelectedUnit();
         if (!target)
         {
-            handler->SendSysMessage(LANG_SELECT_CHAR_OR_CREATURE);
-            handler->SetSentErrorMessage(true);
-            return false;
+            // If no target selected, then target is player.
+            target = handler->GetSession()->GetPlayer();
```

Comment:
Actually this change is totally useless because https://github.com/TrinityCore/TrinityCore/blob/2e586bc520e7eda2387e850c3c0049e6daa880de/src/server/game/Chat/Chat.cpp#L617 already returns the session player if there is no Unit*, did you check how getSelectedUnit() works before modifying the code and debugged what happens when no target is selected ingame ?